art/utils.py

```diff
@@ -630,12 +636,18 @@ def compute_success_array(
     :param batch_size: Batch size.
     :return: Percentage of successful adversarial samples.
     """
-    adv_preds = np.argmax(classifier.predict(x_adv, batch_size=batch_size), axis=1)
+    preds = classifier.predict(x_clean, batch_size=batch_size)
```

Comment:
We should be able to save the computation to evaluate `x_clean` for `targeted` and only evaluate `x_adv`.

Reply:
Committed change to compute only adv_preds first and return attack_success before moving on to clean preds. Not sure if most elegant solution, but should work. Waiting on pytest.